e2e/utils/fixture.ts

```diff
@@ -0,0 +1,277 @@
+import { dirSync } from "tmp";
+
+interface RunCommandOptions {
+  silenceError?: boolean;
+  env?: Record<string, string>;
+  cwd?: string;
+  silent?: boolean;
+}
+
+type RunCommandResult = { stdout: string; stderr: string; combinedOutput: string };
+
+const ORIGIN_GIT = "origin.git";
+const REGISTRY = "http://localhost:4872/";
+
+export const E2E_ROOT = isCI
+  ? dirSync({ prefix: "lerna-e2e-" }).name
+  : joinPathFragments("/tmp", "lerna-e2e");
+
+ensureDirSync(E2E_ROOT);
+
+/**
+ * A initialized Fixture creates an entry within /tmp/lerna-e2e for the given fixture name with the following structure:
+ *
+ * /{{ FIXTURE_NAME }}
+ *   /origin (a bare git repo which acts as the git origin for the lerna workspace under test)
+ *   /lerna-workspace (the lerna workspace under test, created using `lerna init`)
+ *
+ */
+export class Fixture {
+  private readonly fixtureRootPath = joinPathFragments(E2E_ROOT, this.name);
+  private readonly fixtureWorkspacePath = joinPathFragments(this.fixtureRootPath, "lerna-workspace");
+  private readonly fixtureOriginPath = joinPathFragments(this.fixtureRootPath, ORIGIN_GIT);
+
+  constructor(private readonly name: string, private readonly packageManager: "npm" = "npm") {}
+
+  /**
+   * Perform the necessary setup in order to begin testing lerna behavior.
+   *
+   * NOTE: We intentionally do not run `lerna init` within the fixture init, because `lerna init` itself
+   * needs to be transparently tested with its various options configured.
+   */
+  async init(): Promise<void> {
+    this.createFixtureRoot();
+    await this.createGitOrigin();
+    this.createEmptyDirectoryForWorkspace();
+  }
+
+  /**
+   * Tear down the entirety of the fixture including the git origin and the lerna workspace under test.
+   */
+  async destroy(): Promise<void> {
+    await remove(this.fixtureRootPath);
+  }
+
+  /**
+   * Resolve the file path of a given file within the lerna workspace under test.
+   */
+  getWorkspacePath(path?: string): string {
+    return path ? joinPathFragments(this.fixtureWorkspacePath, path) : this.fixtureWorkspacePath;
+  }
+
+  /**
+   * Read the contents of a given file within the lerna workspace under test.
+   */
+  async readWorkspaceFile(file: string): Promise<string> {
+    return readFile(this.getWorkspacePath(file), "utf-8");
+  }
+
+  /**
+   * Execute a given command in the root of the lerna workspace under test within the fixture.
+   * This has been given a terse name to help with readability in the spec files.
+   */
+  async exec(
+    command: string,
+    opts: Exclude<RunCommandOptions, "cwd"> = {
+      silenceError: false,
+      env: undefined,
+    }
+  ): Promise<RunCommandResult> {
+    return this.execImpl(command, { ...opts, cwd: this.fixtureWorkspacePath });
+  }
+
+  /**
+   * Resolve the locally published version of lerna and run the `init` command, with an optionally
+   * provided arguments.
+   */
+  async lernaInit(args?: string): Promise<RunCommandResult> {
```

Comment:
Minor suggestion, but if you change this method signature to `lernaInit(args: string = "")` then the `||` isn't needed in the interpolation below: `...init ${args}`